include/mesh.tcc

```diff
@@ -1075,28 +1075,53 @@ std::vector<Eigen::Matrix<double, 3, 1>>
   return particle_coordinates;
 }
 
+//! Return particle scalar data
+template <unsigned Tdim>
+std::vector<double> mpm::Mesh<Tdim>::particles_scalar_data(
+    const std::string& attribute) {
+  std::vector<double> scalar_data;
+  scalar_data.reserve(particles_.size());
+  // Iterate over particles and add scalar value to data
+  for (auto pitr = particles_.cbegin(); pitr != particles_.cend(); ++pitr)
+    scalar_data.emplace_back((*pitr)->scalar_data(attribute));
+  return scalar_data;
+}
+
+//! Return particle vector data
+template <unsigned Tdim>
+std::vector<Eigen::Matrix<double, 3, 1>> mpm::Mesh<Tdim>::particles_vector_data(
+    const std::string& attribute) {
+  std::vector<Eigen::Matrix<double, 3, 1>> vector_data;
+  // Iterate over particles
+  for (auto pitr = particles_.cbegin(); pitr != particles_.cend(); ++pitr) {
+    Eigen::Matrix<double, 3, 1> data;
+    data.setZero();
+    auto pdata = (*pitr)->vector_data(attribute);
+    // Fill vector_data to the size of dimensions
+    for (unsigned i = 0; i < pdata.size(); ++i) data(i) = pdata(i);
+
+    // Add to a tensor of data
+    vector_data.emplace_back(data);
+  }
+  return vector_data;
+}
+
 //! Return particle tensor data
 template <unsigned Tdim>
 template <unsigned Tsize>
 std::vector<Eigen::Matrix<double, Tsize, 1>>
     mpm::Mesh<Tdim>::particles_tensor_data(const std::string& attribute) {
   std::vector<Eigen::Matrix<double, Tsize, 1>> tensor_data;
-  try {
-    // Iterate over particles
-    for (auto pitr = particles_.cbegin(); pitr != particles_.cend(); ++pitr) {
-      Eigen::Matrix<double, Tsize, 1> data;
-      data.setZero();
-      auto pdata = (*pitr)->tensor_data(attribute);
-      // Fill stresses to the size of dimensions
-      for (unsigned i = 0; i < pdata.size(); ++i) data(i) = pdata(i);
-
-      // Add to a tensor of data
-      tensor_data.emplace_back(data);
-    }
-  } catch (std::exception& exception) {
-    console_->error("{} #{}: {} {}\n", __FILE__, __LINE__, exception.what(),
-                    attribute);
```

Comment:
I don't quite follow the benefit of removing this `catch` and subsequent `error` message @bodhinandach